app/ante/evm/04_validate.go

```diff
@@ -99,13 +98,19 @@ func ValidateTx(tx sdktypes.Tx) (*tx.Fee, error) {
 	return authInfo.Fee, nil
 }
 
+// CheckTxFee checks if the Amount and GasLimit fields of the txFeeInfo input
+// are equal to the txFee coins and the txGasLimit value.
+// The function expects txFeeInfo to contains coins in the original decimal
+// representation.
 func CheckTxFee(txFeeInfo *tx.Fee, txFee sdktypes.Coins, txGasLimit uint64) error {
```

Comment:
NIt: a nice to have would be to refactor this function `txFee` input param  and the `DecoratorUtils` `TxFee` field to be a `big.Int` instead of `sdk.Coins{}`. Because these are the fees on the EVM side, and may not represent the effective fees when using a denom with different precision. Having these as sdk.Coins brings more confusion.

Reply:
Hey @GAtom22, not sure what you mean. Could please take a look at the last commit to see if that it was you were suggesting?